tests/sweep_framework/serialize.py

```diff
@@ -0,0 +1,62 @@
+# SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+from ttnn import *
+import tt_lib
+from statuses import VectorValidity
+
+
+def serialize(object):
+    if "attribute_names" in dir(object) and "attribute_values" in dir(object):
+        serialized_object = dict()
+        serialized_object["type"] = str(type(object)).split("'")[1]
+        attr_names = object.attribute_names
+        attr_values = object.attribute_values()
+        for i in range(len(attr_names)):
+            if attr_names[i] == "shard_spec":
```

Comment:
@cfjchu Demo'd a C++ library that uses reflection called glaze. It serializes C++ types/classes into JSON and back. It was supposed to be added to metal by this week. Gameplan is when that is added we can convert the current implementation to use that since it's more generic. For now, ShardSpec serialization works but it isn't as generic or clean.

Reply:
Update: We have a JSON library tt::stl::json based on the nlohmann library that @arakhmati added. We'll use that.